src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,47 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+    public static final int FIGURE_COUNT = 5;
+
+    public Figure getRandomFigure() {
+        int figureNumber = random.nextInt(FIGURE_COUNT);
+        String color = colorSupplier.getRandomColor();
+
+        switch (figureNumber) {
+            case 0:
+                final double radius = random.nextDouble() * 100;
```

Comment:
Creating a figure, it's better not to pass expressions directly in the constructor for better code readability. Instead, create a variable for the radius and then pass it to the constructor.